src/plots/gl3d/layout/layout_attributes.js

```diff
@@ -140,7 +140,7 @@ module.exports = {
         valType: 'enumerated',
         role: 'info',
         values: ['orbit', 'turntable', 'zoom', 'pan', false],
-        dflt: 'turntable',
+        dflt: 'orbit',
```

Comment:
Hmm. I would prefer removing the `dflt` key entirely - as the default value is now dynamic. 

Could you also add some info in the `description` about the new orbit-vs-turntable logic?

Reply:
Once I tried to remove `dragmode.dflt`, the default behaviour for drag using right-click (orbit/turntable) is not activated.
So we may keep it there.

Reply:
You need to pass the dynamic default to `dragmode` here:
https://github.com/plotly/plotly.js/blob/c9eed54f2fb16f9d2c87d5a7a0fad60c046bccc0/src/plots/gl3d/layout/defaults.js#L141

`getDfltFromLayout` is part of it but then there's the dynamic default based on `camera`. I don't think you want to be *altering* `sceneIn.dragmode` with `enableTurnTable`, you just want to be setting the dynamic default. The logic is something like:

- If `getDfltFromLayout` returns a value, use that.
- If not but the user provides a `camera` that's not `{x: 0, y: 0, z: >0}` `dfltDragmode='orbit'`
- If neither of those, `dfltDragmode = 'turntable'`
- Then you call `coerce('dragmode', dfltDragmode`) so the user can still explicitly provide whatever.
- The only case that's still weird at that point is if the user explicitly says `dragmode: 'turntable', camera: {up: {not z-up}}`. My inclination at that point is to set `sceneOut.camera.up` (and NOT `sceneIn.camera.up`) to `{x: 0, y: 0, z: 1}`. That won't change the behavior - the user might still wonder why `up` is not being honored - but at least it would raise an alarm if you call `Plotly.validate`.

Reply:
Thanks @alexcjohnson, I will modify it to be this way.